drivers: wifi: siwx91x: Add support for link mode query

silabs-MuzaffarA opened the pull request:
Mapped the link mode (wireless mode) from wireless info to the wifi status API.

jerome-pouiller commented on drivers/wifi/siwx91x/siwx91x_wifi.c:
```diff
@@ -73,7 +73,7 @@ int siwx91x_status(const struct device *dev, struct wifi_iface_status *status)
 		sl_wifi_operational_statistics_t operational_statistics = { };
 
-		status->link_mode = WIFI_LINK_MODE_UNKNOWN;
+		status->link_mode = wlan_info.wireless_mode;
```
Don't you need to convert the value? If by chance the values used by Wiseconnect and by Zephyr are the same, you need to mention that in a comment.

muahmed-silabs replied:
Yes, the values are the same. I'll add a comment.
Also, should it be explicitly typecasted?

silabs-MuzaffarA committed:
drivers: wifi: siwx91x: Add support for link mode query

Mapped the link mode (wireless mode) from wireless info to the wifi
status API.

Signed-off-by: Muzaffar Ahmed <muzaffar.ahmed@silabs.com>
